### PR TITLE
Adds the configuration for MetricsScrapeWaitDuration of Druid.

### LINKED ISSUE #648: [Feature] Improvements for ETCD Druid to accommodate compaction job better with the compaction dashboard.
**Feature (What you would like to be added):**
Three short term improvements have been identified in ETCD Druid so that compaction metrics are more meaningful for the compaction dashboard.
- [x]  Compaction Jobs follow a difficult naming convention when they are created by ETCD Druid. The current format is like - `<ETCD UID>-compact-job`. Due to this convention, the pods created by the job have names like `<ETCD UID>-compact-job-<POD UID>`. Due to this the regex expression, which is required to scrape POD resource usage, needs to include `*` infront and afterwards of the string `compact-job`. The regex expression takes the form of `*-compact-job-"`. If we can change the naming convention of compaction job while creating by druid to `compact-job-<ETCD UID`, the regex expression would need only the `*` at the end of the string `compact-job`.
- [x] Currently, we update the metrics `metricsJobCurrent` only when we create a new compaction job in druid and delete the earlier successful/failed compaction job. So, the dashboard project not so accurate completion time for a compaction job. See, we create a new job only when the difference between delta snapshot revision and full snapshot revision cross a certain threshold, configured by end user. It is the same time when we update `metricsJobCurrent` if the last job is completed. To accurately capture  `metricsJobCurrent`, we should actually update it everytime when we check the difference between delta snapshot revision and full snapshot revision. Though we can't be exactly accurate with `metricsJobCurrent` as we are not actively monitoring compaction job. #685 
- [x] Currently, `metricsJobDuration` histogram show average duration of compaction jobs. But the range for average duration of the jobs are captured wrong in the graph of the dashboard. If we follow the attached image, we would know that we don't get a clear idea of the average job duration as all of them falling from 10s to +Inf boundary. The image is also showing unnecessary breakups for milliseconds where no job duration is posted. This is happening because we have not set boundary for the `metricsJobDuration` histogram in ETCD druid. So, Set the expected bucket boundaries for the histogram `metricJobDurationSeconds` so that the compaction job dashboard show proper and meaningful breakup of the duration of the compaction jobs in the Left Y axis of the graph.
      image

- [x] Run `sleep` command for 60 seconds after the compaction job finishes uploading the backup so that even when the upload is finished the fastest, Prometheus gets enough time to capture all network activities.

**Motivation (Why is this needed?):**
To present the compaction job dashboard in more meaningful way to the operators.

**Approach/Hint to the implement solution (optional):**
Make the necessary changes in code for compaction job creation in ETCD Druid.

### PR DESCRIPTION
**Pull request by abdasgupta**
**How to categorize this PR?**

/area monitoring
/kind enhancement

**What this PR does / why we need it**:

This PR makes the provision of MetricsScrapeWaitDuration through bootstrap config of etcd. MetricsScrapeWaitDuration actually supplies --metrics-scrape-wait-duration to compact subcommand of etcdbr command. This duration is needed as prometheus is required to collect necessary metrics for network activity after a fast full snapshot upload at the end of compaction job. Please check https://github.com/gardener/etcd-backup-restore/pull/660 and https://github.com/gardener/etcd-druid/issues/648 for more details.

**Which issue(s) this PR fixes**:
Fixes https://github.com/gardener/etcd-druid/issues/648

**Special notes for your reviewer**:

**Release note**:

```noteworthy  operator
Configure the value for the flag `metrics-scrape-wait-duration` for compaction controller to set a wait duration at the end of every compaction job, to allow for metrics to be scraped by a Prometheus instance.
```

### CODE REVIEW

File: pkg/gardenlet/apis/config/v1alpha1/types.go [hunk 1]
```diff
@@ -596,6 +596,10 @@ type BackupCompactionController struct {
 	// Defaults to 3 hours
 	// +optional
 	ActiveDeadlineDuration *metav1.Duration `json:"activeDeadlineDuration,omitempty"`
+	// MetricsScrapeWaitDuration is the duration to wait for after compaction job is completed, to allow Prometheus metrics to be scraped
+	// Defaults to 60 seconds
```
Comment (timuthy) [inline pkg/gardenlet/apis/config/v1alpha1/types.go:600]: Can you add the required default logic to `pkg/gardenlet/apis/config/v1alpha1/defaults.go`?

File: pkg/gardenlet/apis/config/types.go [hunk 2]
```diff
@@ -491,6 +491,10 @@ type BackupCompactionController struct {
 	// ActiveDeadlineDuration defines duration after which a running backup compaction job will be killed
 	// Defaults to 3 hours
 	ActiveDeadlineDuration *metav1.Duration
+	// MetricsScrapeWaitDuration is the duration to wait for after compaction job is completed, to allow Prometheus metrics to be scraped
+	// Defaults to 60 seconds
+	// +optional
```
Comment (timuthy) [inline pkg/gardenlet/apis/config/types.go]:
```suggestion
```

Not needed for internal API type.

Comment (abdasgupta) [reply pkg/gardenlet/apis/config/types.go]: Done

File: pkg/gardenlet/apis/config/v1alpha1/types.go (hunk 1 above)
Comment (abdasgupta) [reply pkg/gardenlet/apis/config/v1alpha1/types.go:600]: Done

File: pkg/gardenlet/apis/config/v1alpha1/defaults.go [hunk 3]
```diff
@@ -497,4 +497,7 @@ func SetDefaults_ETCDConfig(obj *ETCDConfig) {
 	if obj.BackupCompactionController.EventsThreshold == nil {
 		obj.BackupCompactionController.EventsThreshold = pointer.Int64(1000000)
 	}
+	if obj.BackupCompactionController.MetricsScrapeWaitDuration == nil {
+		obj.BackupCompactionController.MetricsScrapeWaitDuration = &metav1.Duration{Duration: 60 * time.Second}
+	}
```
Comment (timuthy) [inline pkg/gardenlet/apis/config/v1alpha1/defaults.go:502]: Can you please add tests in `defaults_test.go`?

**Review by timuthy — COMMENTED** [review]
Looks good in general. One part is still missing, though.

File: pkg/gardenlet/apis/config/v1alpha1/defaults.go (hunk 3 above)
Comment (abdasgupta) [reply pkg/gardenlet/apis/config/v1alpha1/defaults.go:502]: Added the tests. PTAL.

**Review by timuthy — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**